src/create-remote-resource.js

```diff
@@ -1,184 +1,37 @@
-import { useState, useEffect, useCallback, useRef, useContext } from "react";
-import { createStore } from "redux";
-import { Map as ImmutableMap } from "immutable";
-import Maybe from "data.maybe";
-import Context from "./Context";
-import createTaskManager from "./create-task-manager";
-
-const REGISTER_RESOURCE = "REGISTER_RESOURCE";
-const RECEIVE_DATA = "RECEIVE_DATA";
-const DELETE_ENTRY = "DELETE_ENTRY";
-const SUBSCRIPTION_STARTED = "SUBSCRIPTION_STARTED";
-const SUBSCRIPTION_ENDED = "SUBSCRIPTION_ENDED";
-
-const store = createStore((state = ImmutableMap(), action) => {
-  switch (action.type) {
-    case REGISTER_RESOURCE:
-      return state.set(action.resourceId, ImmutableMap());
-    case RECEIVE_DATA:
-      return state.setIn(
-        [action.resourceId, action.entryKey],
-        ImmutableMap({
-          updatedAt: action.now,
-          data: action.data,
-          hasSubscription: false
-        })
-      );
-    case SUBSCRIPTION_STARTED:
-      return state.setIn(
-        [action.resourceId, action.entryKey, "hasSubscription"],
-        true
-      );
-    case SUBSCRIPTION_ENDED:
-      return state.setIn(
-        [action.resourceId, action.entryKey, "hasSubscription"],
-        false
-      );
-    case DELETE_ENTRY:
-      return state.deleteIn([action.resourceId, action.entryKey]);
-    default:
-      return state;
-  }
-});
-
-const defaultCreateCacheKey = args => args.join("-") || "INDEX";
+import uuid from "uuid/v1";
+import store, { selectEntry, selectResource } from "./store";
 
 const createRemoteResource = ({
-  id: resourceId,
-  load: loader = () => Promise.resolve(),
+  load = () => Promise.resolve(),
   save = () => Promise.resolve(),
   delete: destroy = () => Promise.resolve(),
```

Comment:
Ah, true. Forgot that I made these defaulted.

What do you think would be best:

1. Leaving them defaulted and updating the docs to say they just won't do anything?
2. Removing the defaults and keeping the docs the same?

Reply:
Hrm, I think in this case, leaving the defaults and throwing an error or console.error may be a good way.

```
save: () => {
  console.error("The save action of a resource was called but does not exist on the resource. Resources must be initialized with the save option to use the save action. https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource")
  
  return Promise.resolve();
}
```

Reply:
I thought about using an invariant for these originally, but then I thought, if they're trying to use something that doesn't exist then an error will be thrown anyway. Almost seems unnecessary to throw an error ourselves. Thoughts?

Reply:
True, I was thinking if we could give the user the most helpful error message we can.  Most of the time I assume a user will do a hook in the same component, but if they ever pass down the component, it might get confusing as to why it is undefined. With an error saying it is undefined at least they have a breadcrumb as to why it occurred. Not sure on this one! I think either way will work :-)!

Reply:
Yeah, that's true. I have a hand full of invariants I want to implement already, so I wonder if we should take out the defaults for now, and then try to come up with good, helpful messages at some point.